Wikipedia/Code/NotificationsCenterViewController.swift

```diff
@@ -175,32 +175,49 @@ extension NotificationsCenterViewController: NotificationCenterViewModelDelegate
         }
     }
 
-    /// This will be removed in the final implementation
+    /// TODO: This will be removed in the final implementation
     fileprivate func userDidSwipeCell(indexPath: IndexPath?) {
-        /*
-        guard let indexPath = indexPath, let cellViewModel = viewModel.cellViewModel(indexPath: indexPath) else {
+        guard let indexPath = indexPath,
+              let cellViewModel = viewModel.cellViewModel(indexPath: indexPath) else {
+            return
+        }
+        
+        let swipeActions = cellViewModel.swipeActions(for: viewModel.configuration)
+        guard !swipeActions.isEmpty else {
             return
         }
 
         let alertController = UIAlertController(title: cellViewModel.headerText, message: cellViewModel.bodyText, preferredStyle: .actionSheet)
 
-        let firstAction = UIAlertAction(title: "Action 1", style: .default)
-        let secondAction = UIAlertAction(title: "Action 2", style: .default)
-        let thirdAction = UIAlertAction(title: "Action 3", style: .default)
-        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel)
-
-        alertController.addAction(firstAction)
-        alertController.addAction(secondAction)
-        alertController.addAction(thirdAction)
-        alertController.addAction(cancelAction)
+        swipeActions.forEach { action in
+            
+            let alertAction: UIAlertAction
+            switch action {
+            case .markAsReadOrUnread(let data):
+                alertAction = UIAlertAction(title: data.text, style: .default, handler: { alertAction in
+                    //TODO: Mark as Read handling
+                })
+            case .notificationSubscriptionSettings(let data):
+                alertAction = UIAlertAction(title: data.text, style: .default, handler: { alertAction in
+                    let userActivity = NSUserActivity.wmf_notificationSettings()
+                    NSUserActivity.wmf_navigate(to: userActivity)
+                })
+            case .custom(let data):
+                alertAction = UIAlertAction(title: data.text, style: .default, handler: { alertAction in
+                    let url = data.url
+                    self.navigate(to: url)
+                })
+            }
+            
+            alertController.addAction(alertAction)
+        }
```

Comment:
Currently, after presenting the action sheet, the user can’t dismiss it without selecting an action. We should add a cancel `UIAlertAction` to this action sheet. That’ll allow both tapping `Cancel` to close the sheet and tapping outside the action sheet to dismiss it.

Reply:
@staykids good idea, this is done!

Reply:
@tonisevener 👍🏽 

I think the preferable approach here would actually be using the `.cancel` style and dropping the handler parameter: `let cancelAction = UIAlertAction(title: CommonStrings.cancelActionTitle, style: .cancel)`

This allows the tapping outside of the sheet to dismiss it action, and also logically separates the Cancel button from the primary sheet actions to allow a persistent escape hatch for users as the list of actions grows.

Reply:
The `.cancel` style approach also prevents iPad presentations of the sheet from displaying the extraneous cancel action, as a tap outside the sheet to dismiss it is the popover presentation default behavior.

Reply:
@staykids doh! This is done. Thanks again for cleaning up my sloppiness.